gcc/ChangeLog
2007-11-27  Bernhard Fischer  <>

	* config/arm/arm-protos.h (arm_vector_mode_supported_p,
	arm_hard_regno_mode_ok, const_ok_for_arm): Do not hide non-rtx related
	function prototypes in RTX_CODE.
	* genopinit.c: Include tm_p.h.

--- a/gcc/config/arm/arm-protos.h
+++ b/gcc/config/arm/arm-protos.h
@@ -40,15 +40,14 @@ extern HOST_WIDE_INT thumb_compute_initi
 							       unsigned int);
 extern unsigned int arm_dbx_register_number (unsigned int);
 extern void arm_output_fn_unwind (FILE *, bool);
-  
 
 #ifdef TREE_CODE
 extern int arm_return_in_memory (const_tree);
 #endif
-#ifdef RTX_CODE
 extern bool arm_vector_mode_supported_p (enum machine_mode);
 extern int arm_hard_regno_mode_ok (unsigned int, enum machine_mode);
 extern int const_ok_for_arm (HOST_WIDE_INT);
+#ifdef RTX_CODE
 extern int arm_split_constant (RTX_CODE, enum machine_mode, rtx,
 			       HOST_WIDE_INT, rtx, rtx, int);
 extern RTX_CODE arm_canonicalize_comparison (RTX_CODE, enum machine_mode,
--- a/gcc/genopinit.c
+++ b/gcc/genopinit.c
@@ -487,6 +487,7 @@ from the machine description file `md'.
   printf ("#include \"expr.h\"\n");
   printf ("#include \"optabs.h\"\n");
   printf ("#include \"reload.h\"\n\n");
+  printf ("#include \"tm_p.h\"\n\n");
 
   printf ("void\ninit_all_optabs (void)\n{\n");
 
